src/main/java/com/wultra/app/mobileutilityserver/config/SecurityConfig.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.DelegatingPasswordEncoder;
+import org.springframework.security.crypto.password.MessageDigestPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Security configuration.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true, jsr250Enabled = true)
+public class SecurityConfig {
+
+    @Value("${mobile-utility-server.security.auth.basicHttp.stateless}")
+    private boolean stateless;
+
+    @Value("${mobile-utility-server.security.auth.algorithm}")
+    private String algorithm;
+
+    @Value("${mobile-utility-server.security.auth.bcrypt.cycles}")
+    private int bcryptCycles;
+
+    private final String[] supportedHashAlgorithms = { "SHA-256", "bcrypt" };
+
+    @Bean
+    public UserDetailsService userDetailsService(PasswordEncoder passwordEncoder) {
+        final InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
+        manager.createUser(User.withUsername("admin")
+                .password(passwordEncoder.encode("admin"))
+                .roles("ADMIN")
+                .build());
+        return manager;
+    }
+
+    /**
+     * Configure SHA-256 or bcrypt password encoder. Note that since the passwords are technical, using old SHA-256
+     * algorithm does not cause security issues. Bcrypt is used as default in case no prefix is specified.
+     * See the following URL for constant details:
+     * <a href="https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/crypto/factory/PasswordEncoderFactories.html#createDelegatingPasswordEncoder()">PasswordEncoderFactories.createDelegatingPasswordEncoder()</a>
+     * @return Delegating password encoder.
+     */
+    @Bean
+    @SuppressWarnings("deprecation")
+    public PasswordEncoder passwordEncoder() throws NoSuchAlgorithmException {
+        if (!List.of(supportedHashAlgorithms).contains(algorithm)) {
+            throw new NoSuchAlgorithmException(String.format("Unsupported algorithm specified: %s, must be one of: %s.", algorithm, Arrays.toString(supportedHashAlgorithms)));
+        }
+        final BCryptPasswordEncoder bcrypt = new BCryptPasswordEncoder(bcryptCycles);
+        final MessageDigestPasswordEncoder sha256 = new MessageDigestPasswordEncoder("SHA-256");
+        final Map<String, PasswordEncoder> encoders = new HashMap<>();
+        encoders.put("bcrypt", bcrypt);
+        encoders.put("SHA-256", sha256);
```

Comment:
Since Java 11 we may easily work with immutable collections.

```suggestion
        final Map<String, PasswordEncoder> encoders = Map.of(
                "bcrypt", bcrypt,
                "SHA-256", sha256
        );
```